### PR TITLE
COCICCS-5540:Changes in adding emails to notification list and splunk log issues.

### PR DESCRIPTION
**Pull request by msivatmob**

### CODE REVIEW

File: tvaultapi/src/main/java/com/tmobile/cso/vault/api/controller/OIDCUtil.java [hunk 1]
```diff
@@ -1076,4 +1076,82 @@ private String encodeValue(String value) {
...
+					if (vaulesArray.size() > 0) {
+						Set<String> groupEmailsSet = new HashSet<>();
+						// Adding to set to remove duplicates
+						for (int i=0;i<vaulesArray.size();i++) {
+							JsonObject adObject = vaulesArray.get(i).getAsJsonObject();
+							groupEmailsSet.add(adObject.get("mail").getAsString());
+						}
+						for (String groupEmail: groupEmailsSet) {
+							DirecotryGroupEmail directoryGroupEmail = new DirecotryGroupEmail();
+							directoryGroupEmail.setId(null);
```
Comment (alexrajan3) [inline tvaultapi/src/main/java/com/tmobile/cso/vault/api/controller/OIDCUtil.java:1125]: Why this value is set as null ? we are getting both "Id" and "email" from AAD rt ?

File: tvaultapi/src/main/java/com/tmobile/cso/vault/api/service/OIDCAuthService.java [hunk 2]
```diff
@@ -431,4 +431,21 @@
...
+     * @return
+     */
+    public ResponseEntity<DirectoryObjects> searchGroupEmailInAzureAD(String email) {
+        String ssoToken = oidcUtil.getSSOToken();
+        DirectoryObjects groups = new DirectoryObjects();
+        if (!StringUtils.isEmpty(ssoToken)) {
+            List<DirecotryGroupEmail> allGroups = oidcUtil.getGroupsEmailFromAAD(ssoToken, email);
+
+            DirectoryObjectsList groupsList = new DirectoryObjectsList();
+            groupsList.setValues(allGroups.toArray(new DirecotryGroupEmail[allGroups.size()]));
```
Comment (alexrajan3) [inline tvaultapi/src/main/java/com/tmobile/cso/vault/api/service/OIDCAuthService.java:446]: Null pointer exception could be thrown in allGroups.size()

**Review by alexrajan3 — CHANGES_REQUESTED** [review]
Please add enough Junit test cases to cover the new methods

File: tvaultapi/src/main/java/com/tmobile/cso/vault/api/service/OIDCAuthService.java (hunk 2 above)
Comment (msivatmob) [reply tvaultapi/src/main/java/com/tmobile/cso/vault/api/service/OIDCAuthService.java:446]: checked.

File: tvaultapi/src/main/java/com/tmobile/cso/vault/api/controller/OIDCUtil.java (hunk 1 above)
Comment (msivatmob) [reply tvaultapi/src/main/java/com/tmobile/cso/vault/api/controller/OIDCUtil.java]: the id is not using so we can remove the null value.

File: tvaultapi/src/main/java/com/tmobile/cso/vault/api/controller/OIDCUtil.java [hunk 3]
```diff
@@ -1076,4 +1076,81 @@ private String encodeValue(String value) {
...
 	}
+	/**
+	 * To get groupEmail from AAD.
+	 *
+	 * @param ssoToken
+	 * @param email
+	 * @return
+	 */
+	public List<DirecotryGroupEmail> getGroupsEmailFromAAD(String ssoToken, String email) {
+
```
Comment (nithyal1) [inline tvaultapi/src/main/java/com/tmobile/cso/vault/api/controller/OIDCUtil.java:1087]: Reduce the cognitive complexity to 15

Comment (msivatmob) [reply tvaultapi/src/main/java/com/tmobile/cso/vault/api/controller/OIDCUtil.java:1087]: complexity reduced to 15.

**Review by nithyal1 — APPROVED** [review]

**Review by sarathmohan4 — APPROVED** [review]

**Pull request merged**